src/set/tests.rs

```diff
@@ -583,6 +583,264 @@ fn iter_default() {
     assert_default::<IntoIter<Item>>();
 }
 
+#[test]
+#[allow(deprecated)]
+fn take() {
+    let mut index_set: IndexSet<i32> = IndexSet::new();
+    index_set.insert(10);
+    assert_eq!(index_set.len(), 1);
+
+    let result = index_set.take(&10);
+    assert_eq!(result, Some(10));
+    assert_eq!(index_set.len(), 0);
+
+    let result = index_set.take(&20);
+    assert_eq!(result, None);
+}
+
+#[test]
+fn swap_take() {
+    let mut index_set: IndexSet<i32> = IndexSet::new();
+    index_set.insert(10);
+    index_set.insert(20);
+    index_set.insert(30);
+    index_set.insert(40);
+    assert_eq!(index_set.len(), 4);
+
+    let result = index_set.swap_take(&20);
+    assert_eq!(result, Some(20));
+    assert_eq!(index_set.len(), 3);
+    assert_eq!(index_set.as_slice(), &[10, 40, 30]);
+
+    let result = index_set.swap_take(&50);
+    assert_eq!(result, None);
+}
+
+#[test]
+fn sort_unstable() {
+    let mut index_set: IndexSet<i32> = IndexSet::new();
+    index_set.insert(30);
+    index_set.insert(20);
+    index_set.insert(10);
+
+    index_set.sort_unstable();
+    assert_eq!(index_set.as_slice(), &[10, 20, 30]);
+}
+
+#[test]
+fn try_reserve_exact() {
+    let mut index_set: IndexSet<i32> = IndexSet::new();
+    index_set.insert(10);
+    index_set.insert(20);
+    index_set.insert(30);
+    assert_eq!(index_set.capacity(), 3);
```

Comment:
I think this is too dependent on the growth strategy and load factor, which are implementation details of both hashbrown and this crate combined, but it would be ok if the test calls `shrink_to_fit()` first. We still can't be sure indices table won't have higher capacity, but the entries vec will shrink to the exact capacity, and we report the lesser.

The rest of the test is ok for the same reason -- the vec will grow exactly even if the indices table doesn't.